web_common/web_common/web_common.py

```diff
@@ -98,6 +96,5 @@ async def render_template(
     context['csrf_token'] = csrf_token
 
     response = aiohttp_jinja2.render_template(file, request, context)
-    domain = cookie_domain or deploy_config._domain
-    response.set_cookie('_csrf', csrf_token, domain=domain, secure=True, httponly=True)
+    response.set_cookie('_csrf', csrf_token, secure=True, httponly=True, samesite='strict')
```

Comment:
Hmm. I feel like I'm not a bit confused about our previous conversation.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#strict

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#domaindomain-value

It sounds like Strict is necessary for CSRF to work: otherwise I could put this on an evil website (excuse my totally invented HTML, I didn't bother to look up the right syntax and attribute names) and trick people into cancelling their batches, right? 
```
<form method="POST" url="https://batch.hail.is/batches/123/cancel">
<button>
</form>
```

In particular, do we have any CSRF protection at all right now?

---

I agree we should remove the domain attribute:

> Only the current domain can be set as the value, or a domain of a higher order, unless it is a public suffix. Setting the domain will make the cookie available to it, as well as to all its subdomains.
>
> If omitted, this attribute defaults to the host of the current document URL, not including subdomains.

Reply:
The server compares the cookie value to a value in the request (hidden value in the form), which means that in order to create a malicious form the evil site would need to know the value of the cookie which it can't because it is not the same origin.